extensions/amp-analytics/0.1/requests.js

```diff
@@ -207,18 +237,27 @@ export class RequestHandler {
    */
   fire_() {
     const {
+      requestOriginPromise_: requestOriginPromise,
       baseUrlTemplatePromise_: baseUrlTemplatePromise,
       baseUrlPromise_: baseUrlPromise,
       batchSegmentPromises_: segmentPromises,
     } = this;
     const trigger = /** @type {!JsonObject} */ (this.lastTrigger_);
     this.reset_();
 
-    baseUrlTemplatePromise.then(preUrl => {
+    // preconnect to requestOrigin if available, otherwise baseUrlTemplate
+    const preconnectPromise = requestOriginPromise
+      ? requestOriginPromise
+      : baseUrlTemplatePromise;
+
+    preconnectPromise.then(preUrl => {
       this.preconnect_.url(preUrl, true);
       Promise.all([baseUrlPromise, Promise.all(segmentPromises)]).then(
         results => {
-          const baseUrl = results[0];
+          // prepend requestOrigin if available
+          const baseUrl = requestOriginPromise
+            ? preUrl + results[0]
```

Comment:
Does `preUrl` always has trailing forward slash? So that we can simply add two strings together?

Reply:
Good point, currently I actually remove trailing forward slashes in `requestOrigin` because `requests` in the examples have a leading forward slash.

Do you think we should control how URLs are combined (e.g. adding missing slashes) or leave it up to the user when they define `requests` and `requestOrigin`?

Reply:
Discussed offline, we will use https://developer.mozilla.org/en-US/docs/Web/API/URL/URL to construct the URL.